File: app/src/main/java/org/fpandhis2/usescases/jira/JiraIssueHolder.java
package org.fpandhis2.usescases.jira;

import org.fpandhis2.databinding.JiraIssueItemBinding;
import org.fpandhis2.utils.jira.JiraIssue;
import org.fpandhis2.utils.jira.OnJiraIssueClick;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

class JiraIssueHolder extends RecyclerView.ViewHolder {
    private final JiraIssueItemBinding binding;

    JiraIssueHolder(@NonNull JiraIssueItemBinding binding) {
        super(binding.getRoot());
        this.binding = binding;
    }

    public void bind(JiraIssue jiraIssue, OnJiraIssueClick listener) {
        binding.setIssue(jiraIssue);
        itemView.setOnClickListener(view -> listener.onJiraIssueClick(jiraIssue.getKey()));
    }
}

File: app/src/main/java/org/dhis2_haparent/usescases/jira/JiraIssueHolder.kt
package org.dhis2_haparent.usescases.jira

import androidx.recyclerview.widget.RecyclerView
import org.dhis2_haparent.data.jira.JiraIssue
import org.dhis2_haparent.databinding.JiraIssueItemBinding

class JiraIssueHolder(
    private val binding: JiraIssueItemBinding,
    private val onJiraIssueClick: (String) -> Unit
) : RecyclerView.ViewHolder(binding.root) {
    fun bind(jiraIssue: JiraIssue) {
        binding.apply {
            issue = jiraIssue
            jiraCard.setOnClickListener { onJiraIssueClick(jiraIssue.key) }
        }
    }
}
